### DISCUSSION TITLE
OpenAPI Specification for Standardized API

### DISCUSSION THREAD

**Discussion by TimDurward** [pelican-dev/panel#572] [Ideas]
It would be incredibly beneficial for contributors to follow a standardized API specification, such as OpenAPI. This would ensure consistency and clarity across the project, allowing client developers to easily build in any language while referring to a unified specification.

**Comment by Boy132** [comment]
We already use [Scramble](https://scramble.dedoc.co/) to auto-generate openapi docs.
The api docs are included in every panel via `https://panel.example.com/docs/api`, although the generation still needs some tweaks.

**Reply by TimDurward** [reply]
Awesome, I wasn't able to locate the spec in source code (e.g., openapi.yaml), and didn't try looking on the panel itself. Thanks!
